### PR TITLE
Feature more invoice analyst

### PR DESCRIPTION
**Pull request by aschwartz1**
**What did you change?**
Finish Iteration 3
- Add 2 methods which interact with invoices

**Next Steps**
Start Iteration 4

**Known Issues**

### CODE REVIEW

File: lib/sales_analyst.rb [hunk 1]
```diff
@@ -122,8 +124,20 @@ def invoice_status(status)
...
+  end
+
+  def invoice_total(invoice_id)
+    calculate_invoice_total(invoice_id) if invoice_paid_in_full?(invoice_id)
+  end
+
   private
 
+  def calculate_invoice_total(invoice_id)
+    @invoice_item_repo.invoice_total(invoice_id)
```
Comment (aschwartz1) [inline lib/sales_analyst.rb:138]: Made this helper method solely to get line 132 under the 80 character limit (thanks hound)

File: lib/invoice_item_repository.rb [hunk 2]
```diff
@@ -49,6 +49,14 @@ def find_all_by_invoice_id(invoice_id)
     end
   end
 
+  def invoice_total(invoice_id)
```
Comment (jbranham1) [inline lib/invoice_item_repository.rb:52]: Very nice

File: lib/sales_analyst.rb [hunk 3]
```diff
@@ -122,8 +124,20 @@ def invoice_status(status)
...
+    @transaction_repo.paid_in_full?(invoice_id)
+  end
+
+  def invoice_total(invoice_id)
+    calculate_invoice_total(invoice_id) if invoice_paid_in_full?(invoice_id)
+  end
+
   private
 
+  def calculate_invoice_total(invoice_id)
```
Comment (jbranham1) [inline lib/sales_analyst.rb:137]: Love how clean these methods look

**Review by jbranham1 — APPROVED** [review]
DONE with iteration 3!

**Pull request merged**